### ISSUE TITLE
Have we convert imgs to 8bit?

### ISSUE THREAD

**Issue by yt605155624** [hienpham15/CNN_model_ColorConstancy#3]
@hienpham15  
Hello,I have checked your code,I want to ask you that do we must convert the img to 8bit? have you test which will be better to trian  8bit or12/16bit,and if we convert img to 8bit ,can we use the label directly(by nomorlization) or we have to preprocess them? will it influence if I train on 16bit imgs and test on 8 bit normal imgs?

**Comment by hienpham15** [comment]
@yt605155624 
To answer your big question, it depends: 
For the purpose of display: yes, because most typical desktop displays only support 8 bits of color per channel. Therefore, if you want to observe your results (including Color-casted images and Corrected image) then you must perform the conversion.
For the purpose of image processing: no, an image with 12/16-bit per channel always contains more information but also more noisy data than the 8-bit one, so if you want to train with 16-bit images be careful of overfitting problem (and also more computer resources required for training). As in my part, I haven't trained with the 12/16-bit images due to lack of computing resources.
About the labels, here I will assume that you mean the ground truth illuminants corresponding to each image, and in that case, the answer is yes. For example, you have an 8-bit image of size (32x32x3), then the corresponding labels (ground truth illuminant) will be (1x3). 
For the final question, as I have mentioned above, 16-bit images yield more data but also more noises, whether it results in better or worse performance, you will have to see it for yourself.
